### PR TITLE
remove dropout from fluid

### PR DESCRIPTION
**Pull request by ccrrong**
### PR types

Others

### PR changes

APIs

### Describe

To clean the code in paddle.fluid, the useless codes should be removed and code still be used should move to a new position.

For function paddle.fluid.layers.nn.dropout, since it is not used in version 2.x, should be removed.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/dygraph_to_static/transformer_dygraph_model.py [hunk 1]
```diff
@@ -155,7 +156,7 @@ def forward(self, queries, keys, values, attn_bias, cache=None):
             product += attn_bias
         weights = paddle.nn.functional.softmax(product)
         if self.dropout_rate:
-            weights = layers.dropout(weights, dropout_prob=self.dropout_rate)
+            weights = paddle.nn.functional.dropout(weights, p=self.dropout_rate)
             out = layers.matmul(weights, v)
```
Comment (phlrain) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/transformer_dygraph_model.py]: paddle.nn.functional.dropout(weights, p=self.dropout_rate， training=self.training)

File: python/paddle/fluid/tests/unittests/dygraph_to_static/transformer_dygraph_model.py [hunk 2]
```diff
@@ -72,9 +72,10 @@ def __init__(self, process_cmd, d_model, dropout_rate):
                 )
             elif cmd == "d":  # add dropout
                 if dropout_rate:
-                    self.functors.append(
-                        lambda x: layers.dropout(x, dropout_prob=dropout_rate)
+                    self.dropout = paddle.nn.Dropout(
+                        p=dropout_rate, mode="downscale_in_infer"
```
Comment (phlrain) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/transformer_dygraph_model.py:77]: paddle.nn.functional.dropout(weights, p=self.dropout_rate， training=self.training)

**Review by heavengate — APPROVED** [review]

**Review by jzhang533 — APPROVED** [review]
LGTM for fluid API removing

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
